### PR TITLE
chore: update most dependencies

### PR DESCRIPTION
**Pull request by RobinCsl**
**chore: update most dependencies**

and fix resulting issues.

---

**chore: replace eslint-plugin-prettier with just prettier**

---

### CODE REVIEW

File: src/getClientToken.ts [hunk 1]
```diff
@@ -1,22 +1,19 @@
-import { DesktopIAPOptions } from "./types";
+/// <reference lib="dom" />
```
Comment (oreqizer) [inline src/getClientToken.ts:1]: why's this needed?

File: src/authenticationMiddleware.test.ts [hunk 2]
```diff
@@ -11,80 +16,81 @@ test("required returns correct values", (t) => {
...
-  });
+  }
 });
 
 test("validateAudience passes on valid values", (t) => {
   const tests: string[] = [
     "/projects/000000000000/global/backendServices/0000000000000000000",
     "/projects/000000000000/apps/my-sample-project-191923",
   ];
-  tests.forEach(audience => {
```
Comment (oreqizer) [inline src/authenticationMiddleware.test.ts:25]: why not `.forEach`?

**Review by oreqizer — COMMENTED** [review]
two minor questions, otherwise looks okayge

File: src/getClientToken.ts (hunk 1 above)
Comment (RobinCsl) [reply src/getClientToken.ts:1]: This is leftovers of something I tried, thank you for catching it, I'll remove it.

File: src/authenticationMiddleware.test.ts (hunk 2 above)
Comment (RobinCsl) [reply src/authenticationMiddleware.test.ts:25]: I think I had to do this when I tried to update `ava`, but it's no longer part of this PR, so I reverted the change. Thank you.

**Comment by oreqizer** [comment]
approved 👌

**Review by oreqizer — APPROVED** [review]

**Pull request merged**